debugger/src/debugger/configuration.rs

```diff
@@ -0,0 +1,228 @@
+    pub(crate) protocol: Option<WireProtocol>,
+
+    ///Allow the session to erase all memory of the chip or reset it to factory default.
+    #[serde(default)]
+    pub(crate) allow_erase_all: bool,
+
+    /// Flashing configuration
+    pub(crate) flashing_config: FlashingConfig,
+
+    /// Every core on the target has certain configuration.
+    ///
+    /// NOTE: Although we allow specifying multiple core configurations, this is a work in progress, and probe-rs-debugger currently only supports debugging a single core.
+    pub(crate) core_configs: Vec<CoreConfig>,
+}
+
+impl SessionConfig {
+    /// Ensure all file names are correctly specified and that the files they point to are accessible.
+    pub(crate) fn validate_config_files(&mut self) -> Result<(), DebuggerError> {
+        // Update the `cwd`.
+        self.cwd = self.validate_and_update_cwd()?;
+
+        for target_core_config in &mut self.core_configs {
+            // Update the `program_binary` and validate that the file exists.
+            target_core_config.program_binary = match qualify_and_update_os_file_path(
+                self.cwd.clone(),
+                target_core_config.program_binary.as_ref(),
+            ) {
+                Ok(program_binary) => {
+                    if !program_binary.is_file() {
+                        return Err(DebuggerError::Other(anyhow!(
+                            "Invalid program binary file specified '{:?}'",
+                            program_binary
+                        )));
+                    }
+                    Some(program_binary)
+                }
+                Err(error) => {
+                    return Err(DebuggerError::Other(anyhow!(
+                            "Please use the `program-binary` option to specify an executable for this target core. {:?}", error
+                        )));
+                }
+            };
+            // Update the `svd_file` and validate that the file exists.
+            // If there is a problem with this file, warn the user and continue with the session.
+            target_core_config.svd_file = match qualify_and_update_os_file_path(
+                self.cwd.clone(),
+                target_core_config.svd_file.as_ref(),
+            ) {
+                Ok(svd_file) => {
+                    if !svd_file.is_file() {
+                        log::error!("SVD file {:?} not found.", svd_file);
+                        None
+                    } else {
+                        Some(svd_file)
+                    }
+                }
+                Err(error) => {
+                    // SVD file is not mandatory.
+                    log::debug!("SVD file not specified: {:?}", &error);
+                    None
+                }
+            };
+        }
+
+        Ok(())
+    }
+
+    /// Validate the new cwd, or else set it from the environment.
+    pub(crate) fn validate_and_update_cwd(&self) -> Result<Option<PathBuf>, DebuggerError> {
```

Comment:
I think this function name does not really tell what it does.

Reply:
Done. Also updated the inline docs.